store/validator.go

```diff
@@ -51,6 +51,14 @@ func (vs *validatorStore) hasValidator(addr crypto.Address) bool {
 	return ok
 }
 
+func (vs *validatorStore) ValidatorAddresses() []crypto.Address {
+	validatorAddresses := make([]crypto.Address, 0, len(vs.numberMap))
```

Comment:
Here, better we use `addressMap`, It should be faster, since we don't need to calculate the address from the public key of the validator. The validator's address is the map key:

```go
for addr, _ := range vs.addresMap {
...
	}
```

Suggestion:
I recommend to rename `validatorAddresses` to `addrs`. This is a local variable within a short scope, a shorter name could be more readable.